pkg/collection/entry/metadata.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package entry
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// Metadata provides mime type and filename to file entry.
+type Metadata struct {
+	MimeType string
+	Filename string
+}
+
+// NewMetadata creates a new Metadata.
+func NewMetadata(fileName string) *Metadata {
+	return &Metadata{
+		Filename: fileName,
+	}
+}
+
+// WithMimeType adds mime type to Metadata.
+func (m *Metadata) WithMimeType(mimeType string) *Metadata {
```

Comment:
if `MimeType` is already exported, why do you need this function for?

Reply:
I do not think that we need this method, just set mime type in the constructor.